### PR TITLE
Removed dom4j dependency from core Errai

### PR DESCRIPTION
**Pull request by hrstoyanov**
Max,
Trying to get rid of the dom4j and use the jdk built-in XML  dom parser. This is only for the core Errai . I think reflections is the only one using dom4j. The demos are not touched and some of them still use dom4j.

Dom4j is also used by Hibernate and I thought the Errai stopped using hibernate, but I just noticed that the errai-jpa is back?

### CODE REVIEW

**Comment by mbarkley** [comment]
ok to test

**Comment by mbarkley** [comment]
@hrstoyanov errai-jpa is deprecated but not removed yet. It will remain in Errai 4.x, but probably not in Errai 5.

**Comment by hrstoyanov** [comment]
Thanks Max,
Can you help me understand which tests fail with my patch. I do see some failures with ```Codegen```, but seems not related to ```Reflections```?

```
[INFO] Running org.jboss.errai.codegen.test.AllStatementBuilderTests
package org.foo;

import java.io.Serializable;

public class Bar implements Serializable {
  private String name;
}
[ERROR] Tests run: 247, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.124 s <<< FAILURE! - in org.jboss.errai.codegen.test.AllStatementBuilderTests
[ERROR] testEncodeAnnotation(org.jboss.errai.codegen.test.AnnotationEncoderTest)  Time elapsed: 0.002 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<...tation.Target(value=[[METHOD]])"; } public java.la...> but was:<...tation.Target(value=[{METHOD}])"; } public java.la...>
	at org.jboss.errai.codegen.test.AnnotationEncoderTest.testEncodeAnnotation(AnnotationEncoderTest.java:38)

[ERROR] testEncodeAnnotationWithMultipleProperties(org.jboss.errai.codegen.test.AnnotationEncoderTest)  Time elapsed: 0 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<...yTestAnnotation(foo=[barfoo], testEum=FOURTH)"; ...> but was:<...yTestAnnotation(foo=[\"barfoo\"], testEum=FOURTH)"; ...>
	at org.jboss.errai.codegen.test.AnnotationEncoderTest.testEncodeAnnotationWithMultipleProperties(AnnotationEncoderTest.java:56)

[INFO] Running org.jboss.errai.codegen.test.IfBlockBuilderTest
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in org.jboss.errai.codegen.test.IfBlockBuilderTest
[INFO] Running org.jboss.errai.codegen.test.StatementBuilderTest
[INFO] Tests run: 51, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.011 s - in org.jboss.errai.codegen.test.StatementBuilderTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR] org.jboss.errai.codegen.test.AnnotationEncoderTest.testEncodeAnnotation(org.jboss.errai.codegen.test.AnnotationEncoderTest)
[ERROR]   Run 1: AnnotationEncoderTest.testEncodeAnnotation:38->AbstractCodegenTest.assertEquals:43 expected:<...tation.Target(value=[[METHOD]])"; } public java.la...> but was:<...tation.Target(value=[{METHOD}])"; } public java.la...>
[ERROR]   Run 2: AnnotationEncoderTest.testEncodeAnnotation:38->AbstractCodegenTest.assertEquals:43 expected:<...tation.Target(value=[[METHOD]])"; } public java.la...> but was:<...tation.Target(value=[{METHOD}])"; } public java.la...>
[INFO] 
[ERROR] org.jboss.errai.codegen.test.AnnotationEncoderTest.testEncodeAnnotationWithMultipleProperties(org.jboss.errai.codegen.test.AnnotationEncoderTest)
[ERROR]   Run 1: AnnotationEncoderTest.testEncodeAnnotationWithMultipleProperties:56->AbstractCodegenTest.assertEquals:43 expected:<...yTestAnnotation(foo=[barfoo], testEum=FOURTH)"; ...> but was:<...yTestAnnotation(foo=[\"barfoo\"], testEum=FOURTH)"; ...>
[ERROR]   Run 2: AnnotationEncoderTest.testEncodeAnnotationWithMultipleProperties:56->AbstractCodegenTest.assertEquals:43 expected:<...yTestAnnotation(foo=[barfoo], testEum=FOURTH)"; ...> but was:<...yTestAnnotation(foo=[\"barfoo\"], testEum=FOURTH)"; ...>
[INFO] 
[INFO] 
[ERROR] Tests run: 554, Failures: 2, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Errai Internal BOM (Bill Of Materials) ............. SUCCESS [  0.001 s]
[INFO] JBoss Errai BOM .................................... SUCCESS [  0.886 s]
[INFO] Errai .............................................. SUCCESS [  0.261 s]
[INFO] Reflections parent ................................. SUCCESS [  0.267 s]
[INFO] Reflections ........................................ SUCCESS [  0.271 s]
[INFO] Errai::Common ...................................... SUCCESS [  2.681 s]
[INFO] Errai::CDI::EE6 API Overrides ...................... SUCCESS [  0.078 s]
[INFO] Errai::Codegen ..................................... FAILURE [  1.987 s]
```

**Comment by mbarkley** [comment]
Where are you seeing those test failures? On Jenkins it is `org.jboss.errai.reflections.ReflecitonTest` that is breaking.

**Comment by hrstoyanov** [comment]
Max, fixed the issues and now the ReflectionTest passes cleanly. Are you able to see the pull request?
... Forgot to squish the commits

**Comment by hrstoyanov** [comment]
```Build finished. 2656 tests run, 5 skipped, 0 failed. ``` ... so why does it marked the tests as failed? Also, clicking "Details" link goes nowhere.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

File: errai-bus/src/main/java/org/jboss/errai/bus/server/servlet/StandardAsyncServlet.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
  * An implementation of {@link AbstractErraiServlet} leveraging asynchronous support of Servlet 3.0.
  *
  * @author Christian Sadilek <csadilek@redhat.com>
- * @author Mike Brock
+ * @author Mike BrockReflectionTests
```
Comment (mbarkley) [inline errai-bus/src/main/java/org/jboss/errai/bus/server/servlet/StandardAsyncServlet.java]: Not sure what happened here, but can you please remove this change?

Comment (hrstoyanov) [reply errai-bus/src/main/java/org/jboss/errai/bus/server/servlet/StandardAsyncServlet.java]: fat fingered :-)

**Comment by hrstoyanov** [comment]
Max,
Fixed the comment on ```StandardAsyncServlet```, I must have fat-fingered it:-)

**Pull request merged**